snow_incident/snow_incident.py

```diff
@@ -0,0 +1,199 @@
+    """
+    data = {"priority": 2}
+
+    try:
+        data["short_description"] = event["Records"][0]["Sns"]["Subject"]
+        data["description"] = event["Records"][0]["Sns"]["Message"]
+        data["timestamp"] = event["Records"][0]["Sns"]["Timestamp"]
+    except KeyError:
+        missing = [
+            i
+            for i in ["Subject", "Message", "Timestamp"]
+            if i not in event["Records"][0]["Sns"]
+        ]
+        raise KeyError(
+            'Malformed message: {"Records": [ {"Sns": {}: missing one of Subject, Message, Timestamp'
+        )
+
+    # If message is JSON, parse for additional information
+    try:
+        jmessage = json.loads(data["description"])
+    except json.decoder.JSONDecodeError:
+        jmessage = {}
+
+    # Extract priority from subject or JSON if present
+    m = re.search(r"\[p([0-5])\]", data["short_description"], re.I)
+    if m:
+        data["priority"] = int(m[1])
+
+    if "priority" in jmessage:
+        try:
+            data["priority"] = int(jmessage["priority"])
+            if data["priority"] < 0 or data["priority"] > 5:
+                raise ValueError
+        except ValueError:
+            raise ValueError(
+                f"Invalid priority {jmessage['priority']}: Must be int between 0 and 5"
+            )
+
+    return data
+
+
+def create_body(
+    default_body: dict,
+    short_description: str,
+    description: str,
+    impact: int = 2,
+    urgency: int = 2,
+    priority: int = 2,
+):
+    """
+    Merge defaults with parsed event elements.
+
+    default_body      - Set of defaults including caller_id, u_category, and u_item keys
+    short_description - Summary
+    description       - Multiline detail of incident
+    impact            - 0 (highest) to 5 (lowest) (Default: 2)
+    urgency           - 0 (highest) to 5 (lowest) (Default: 2)
+    priority          - 0 (highest) to 5 (lowest) (Default: 2)
+    """
+    body = default_body.copy()
+
+    body.update(
+        {
+            "short_description": short_description,
+            "description": description,
+            "impact": impact,
+            "urgency": urgency,
+            "priority": priority,
+        }
+    )
+
+    return body
+
+
+def create_incident(url: str, auth: Any, body: dict) -> str:
+    headers = urllib3.make_headers(basic_auth=":".join(auth))
+    headers.update({"Content-Type": "application/json", "Accept": "application/json"})
+
+    encoded_body = json.dumps(body).encode("utf-8")
+
+    http = urllib3.PoolManager()
+
+    response = http.request("POST", url, headers=headers, body=encoded_body)
+
+    if response.status != 201:
+        full_error = (
+            f"SNOW request failed with status {response.status_code}, ",
+            f"Headers: {response.headers}, ",
+            f"Response: {response.data.decode('utf-8')}",
+        )
+        raise ValueError(full_error)
+
+    return json.loads(response.data.decode("utf-8"))
+
+    return data["number"]
```

Comment:
`data` isn't defined in this context, and also this `return` statement won't be called.